client/src/reducers/annotations.ts

```diff
@@ -29,16 +29,27 @@ const Annotations = (
     labelEditable: { category: null, label: null },
     promptForFilename: true,
   },
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  action: any
+  action: {
+    type: string;
+    data: string;
+    metadataField: string;
+    categoryIndex: number;
+    config: {
+      parameters: {
+        "annotations-data-collection-name": string;
+        user_annotation_collection_name_enabled: boolean;
+        annotations_genesets_name_is_read_only: boolean /* FIXME typescript  */;
+      };
+    };
+  }
 ) => {
   switch (action.type) {
     case "configuration load complete": {
       const dataCollectionName =
         action.config.parameters?.["annotations-data-collection-name"] ?? null;
       const dataCollectionNameIsReadOnly =
         (action.config.parameters?.[
-          "annotations-data-collection-is-read-only"
+          "annotations-data-collection-is-read-only" /* FIXME typescript */
```

Comment:
This line has the following error:

Element implicitly has an 'any' type because expression of type '"annotations-data-collection-is-read-only"' can't be used to index type '{ "annotations-data-collection-name": string; user_annotation_collection_name_enabled: boolean; annotations_genesets_name_is_read_only: boolean; }'.
  Property 'annotations-data-collection-is-read-only' does not exist on type '{ "annotations-data-collection-name": string; user_annotation_collection_name_enabled: boolean; annotations_genesets_name_is_read_only: boolean; }'.ts(7053)

Reply:
It's `-` vs` _`